Python/Product/Cookiecutter/ViewModel/CookiecutterViewModel.cs

```diff
@@ -0,0 +1,749 @@
+            }
+
+            set {
+                if (value != _isLoadingError) {
+                    _isLoadingError = value;
+                    PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(IsLoadingError)));
+                }
+            }
+        }
+
+        public bool IsCreating {
+            get {
+                return _isCreating;
+            }
+
+            set {
+                if (value != _isCreating) {
+                    _isCreating = value;
+                    PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(IsCreating)));
+                }
+            }
+        }
+
+        public bool IsCreatingSuccess {
+            get {
+                return _isCreatingSuccess;
+            }
+
+            set {
+                if (value != _isCreatingSuccess) {
+                    _isCreatingSuccess = value;
+                    PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(IsCreatingSuccess)));
+                }
+            }
+        }
+
+        public bool IsCreatingError {
+            get {
+                return _isCreatingError;
+            }
+
+            set {
+                if (value != _isCreatingError) {
+                    _isCreatingError = value;
+                    PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(IsCreatingError)));
+                }
+            }
+        }
+
+        public TemplateViewModel SelectedTemplate {
+            get {
+                return _selectedTemplate;
+            }
+
+            set {
+                if (value != _selectedTemplate) {
+                    _selectedTemplate = value;
+                    PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(SelectedTemplate)));
+                }
+            }
+        }
+
+        public bool IsOutputFolderEmpty() {
+            if (Directory.Exists(OutputFolderPath)) {
+                var files = Directory.EnumerateFileSystemEntries(OutputFolderPath);
+                return files.Count() == 0;
+            }
+
+            return true;
+        }
+
+        public static string GetUserConfigPath() {
+            var userConfigFilePath = Environment.GetEnvironmentVariable(ConfigEnvironmentVariableName);
+            if (string.IsNullOrEmpty(userConfigFilePath)) {
+                userConfigFilePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), DefaultConfigFileName);
+            }
+
+            return userConfigFilePath;
+        }
+
+        public async Task SearchAsync() {
+            _templateRefreshCancelTokenSource?.Cancel();
+            _templateRefreshCancelTokenSource = new CancellationTokenSource();
+            try {
+                await RefreshTemplatesAsync(SearchTerm, _templateRefreshCancelTokenSource.Token);
+            } catch (OperationCanceledException) {
+            }
+            _templateRefreshCancelTokenSource = null;
```

Comment:
>_templateRefreshCancelTokenSource [](start = 12, length = 33)

This is a new race condition here because it's after the await. I'd just leave the value there (or use Interlocked.CompareExchange, but since this function is always called from the UI thread that isn't necessary, and CTS is a cheap class)